lib/config/folderconfiguration.go

```diff
@@ -275,3 +275,7 @@ func (l FolderDeviceConfigurationList) Len() int {
 func (f *FolderConfiguration) CheckFreeSpace() (err error) {
 	return checkFreeSpace(f.MinDiskFree, f.Filesystem())
 }
+
+func (f *FolderConfiguration) SetFilesystem(fs fs.Filesystem) {
```

Comment:
This method should not exist imho, there is no reason for someone from the outside to be able to manipulate the cache structure

Reply:
How would you test, that the limiter is in place? As we dont have separation between concerns I can not replacing scanning with a dummy, so I need to check some other places. I could not find another "good" place.

This unveils the model is hard to test.